### PR TITLE
[DOC] Add X-Pack links to Elasticsearch Reference

### PR DESCRIPTION
**Pull request by lcawl**
This pull request adds the infrastructure necessary to reference X-Pack content.

It uses a layout similar to what was worked out in the Kibana Reference for  https://github.com/elastic/kibana/pull/12062

It also adds some attribute definitions that are specific to X-Pack and changes the name of the attribute for the X-Pack Reference URL so that it does not cause confusion.

### CODE REVIEW

File: docs/Versions.asciidoc [hunk 1]
```diff
@@ -16,7 +16,7 @@ release-state can be: released | prerelease | unreleased
 :painless:        https://www.elastic.co/guide/en/elasticsearch/painless/master
 :plugins:         https://www.elastic.co/guide/en/elasticsearch/plugins/{branch}
 :javaclient:      https://www.elastic.co/guide/en/elasticsearch/client/java-api/{branch}
-:xpack:           https://www.elastic.co/guide/en/x-pack/{branch}
+:xpack-ref:           https://www.elastic.co/guide/en/x-pack/{branch}
```
Comment (clintongormley) [inline docs/Versions.asciidoc]: Please line up the URL with the ones before and after.  You don't use tabs by any chance, do you?  We have a build check that will cause tabs to fail the build.

File: docs/Versions.asciidoc [hunk 2]
```diff
@@ -25,6 +25,13 @@ release-state can be: released | prerelease | unreleased
 :docker-image:    docker.elastic.co/elasticsearch/elasticsearch:{version}
 :plugin_url:      https://artifacts.elastic.co/downloads/elasticsearch-plugins
 
+:xpack:           X-Pack
+:xpackml:         X-Pack machine learning
+:ml:              machine learning
+:es:              Elasticsearch
+:kib:             Kibana
+
+:xes-repo-dir: {docdir}/../../elasticsearch-extra/x-pack-elasticsearch/docs/en
```
Comment (clintongormley) [inline docs/Versions.asciidoc]: Line the path up with the settings above

**Review by clintongormley — APPROVED** [review]
LGTM (minor alignment issues)

File: docs/Versions.asciidoc (hunk 1 above)
Comment (lcawl) [inline docs/Versions.asciidoc]: Thanks, I'll avoid tabs in the future!

**Pull request merged**